core/src/auctions.ts

```diff
@@ -108,24 +108,34 @@ export const fetchAllInitialAuctions = async function (network: string): Promise
 
 export const fetchAllAuctions = async function (network: string): Promise<AuctionTransaction[]> {
     const auctions = await fetchAllInitialAuctions(network);
+    return await Promise.all(auctions.map(a => enrichAuction(network, a)));
+};
 
+export const enrichAuction = async function (
+    network: string,
+    auction: AuctionInitialInfo
+): Promise<AuctionTransaction> {
     // enrich them with statuses
-    const auctionsWithStatusesPromises = auctions.map((auction: AuctionInitialInfo) =>
-        enrichAuctionWithActualNumbers(network, auction)
-    );
-    const auctionsWithStatuses = await Promise.all(auctionsWithStatusesPromises);
+    const auctionWithStatus = await enrichAuctionWithActualNumbers(network, auction);
 
     // enrich them with price drop
-    const auctionsWithPriceDrop = await Promise.all(auctionsWithStatuses.map(enrichAuctionWithPriceDrop));
+    const auctionWithPriceDrop = await enrichAuctionWithPriceDrop(auctionWithStatus);
 
     // enrich them with market values
-    const auctionsWithMarketValue = await Promise.all(
-        auctionsWithPriceDrop.map(a => enrichAuctionWithMarketValues(a, network))
-    );
+    const auctionWithMarketValue = await enrichAuctionWithMarketValues(auctionWithPriceDrop, network);
 
     // enrich with profit and fee calculation
     const fees = await getApproximateTransactionFees(network);
-    return await Promise.all(auctionsWithMarketValue.map(a => enrichAuctionWithTransactionFees(a, fees)));
+    const auctionWithFees = await enrichAuctionWithTransactionFees(auctionWithMarketValue, fees, network);
+
+    if (auction.debtDAI.toNumber() === 0) {
```

Comment:
I think it's safer to call native bignumber method than convert it to number:
```suggestion
    if (auction.debtDAI.isEqualTo(0)) {
```